src/design-pages/passkey_authentication.rst

```diff
@@ -0,0 +1,318 @@
+Computers' utility, or any other attribute in case of another LDAP server.
+
+LDAP attributes
+***************
+Newly added IPA ``ipapasskey`` LDAP attribute will be added to
+``src/providers/ipa/ipa_opts.c`` to retrieve the passkey mapping data
+string as part of the user search in the backend. Store this in the sysdb. The
+passkey mapping is obtained during the registration phase. Example:
+
+* ipapasskey: passkey:9S87qLk8/RxYJ3skwwYduomAM+/HDtz41N0+w/vRL6aGKJkLMsg+2OhO0E8pK5DuO1KmdK61K8PmH7jiYuOqbg==,9YE1s/f7J47h2A/DXCVFWulqoBXFzCSxcbGEBadkpSUFjwUudhPLnPUTv2qNamakXJgRYCZQ7vpS/t5zXMLnkw==
+
+In the Active Directory case, the
+`altSecurityIdentities <https://learn.microsoft.com/en-us/windows/win32/adschema/a-altsecurityidentities>`__
+LDAP attribute will be used to store the passkey mapping data. Example:
+
+* altSecurityIdentities: passkey:9S87qLk8/RxYJ3skwwYduomAM+/HDtz41N0+w/vRL6aGKJkLMsg+2OhO0E8pK5DuO1KmdK61K8PmH7jiYuOqbg==,9YE1s/f7J47h2A/DXCVFWulqoBXFzCSxcbGEBadkpSUFjwUudhPLnPUTv2qNamakXJgRYCZQ7vpS/t5zXMLnkw==
+
+Finally, newly added IPA ``iparequireuserverification`` LDAP attribute will be
+retrieved from the IPA passkey configuration. Store this in the sysdb
+associated with the IPA domain. This attribute stores a string (on, off,
+default). Example:
+
+* iparequireuserverification: on
+
+User authentication with the key
+********************************
+The idea is to create a helper process similar to the p11_child. The helper
+process will receive the passkey’s user key mapping data as an argument
+alongside other parameters (check
+:ref:`Configuration options<configuration-options>` for more information).
+
+This helper process will heavily rely on the libfido2 library. The first steps
+are to initialize the library and get the list of keys connected to the system.
+Next the helper process will load the configuration parameters in the assert
+structure, that comprises a collection of statements, each statement has a
+mapping for a challenge, a credential, a signature, and ancillary attributes.
+Finally, we will open the key device and ask it for an authentication according
+to the information provided in the
+`assert <https://developers.yubico.com/libfido2/Manuals/fido_assert_new.html>`__
+structure.
+
+Call the passkey child in the PAM responder
++++++++++++++++++++++++++++++++++++++++++++
+In ``pam_forwarder()`` check if passkey authentication is enabled (if
+pam_passkey_auth boolean option is true and pam cmd == SSS_PAM_AUTHENTICATE).
+If so, lookup the sysdb user object of the user logging in and read the sysdb
+passkey mapping value. Parse out the key-handle and public-key strings from the
+passkey mapping value. Read passkey related configuration options
+(passkey_verification, passkey_child_timeout, debug_libfido2) and set up the
+arguments (Domain, Key-handle and Public-key) for the passkey-child.
+
+Execute the passkey_child passing the arguments listed above. The passkey PIN
```

Comment:
It is unclear under what user the passkey_child is launched. Since passkeys often require access to USB or other external bus to communicate, it would be good to mention how to achieve that (udev rules). Specifically, this should be covered if passkey_child does not run as root.

Reply:
I explain this at the top of the document, where I explain how to use the `sssctl` command for the first time.